presto-hive/src/main/java/com/facebook/presto/hive/HiveColumnHandle.java

```diff
@@ -33,6 +35,15 @@
         implements ColumnHandle
 {
     public static final String SAMPLE_WEIGHT_COLUMN_NAME = "__presto__sample_weight__";
+
+    // columns with column index < -10 will be hidden
+    public static final int HIDDEN_COLUMN_INDEX_THRESHOLD = -10;
+
+    public static final int INPUT_FILE_NAME_COLUMN_INDEX = -11;
+    public static final String INPUT_FILE_NAME_COLUMN_NAME = "input__file__name";
```

Comment:
Are we sure we want this to be the name? This is an entirely plausible name for a column a regular table might have. I'd pick something that starts with "$" (like we do with raptor's hidden columns)